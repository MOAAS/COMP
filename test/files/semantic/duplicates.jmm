import io.println(int);
import io.println(int, int) void;
import io.println(int, int, int) void;
import io.println(int, int, boolean) void;
import io.println(int) boolean;
import io.println(int);
import io.println(int) int;

import Quicksort;
import Quicksort;

import static Package;
import static Package(int, boolean);

import Package int;
import Package(int, boolean) int;

class Simple {
    public static void main(String[] arg){
        Simple simple;
        simple = new Simple();
        simple.add(1, 2);
    }

    public int add(int a, int b) {
        return a + b;
    }

    public int add(int a, boolean a, int[] a) {
        return a + b;
    }

    public int add(int a, boolean a, int[] a) {
        return a + b;
    }

    public boolean add(int a, int b, int c) {
        return a + b;
    }

    public int add(int a, int b) {
        return a + b;
    }

    public boolean add(int a, int b) {
        return a + b;
    }

    public int[] add(int a, int b) {
        return a + b;
    }
 }