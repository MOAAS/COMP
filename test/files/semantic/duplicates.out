Warning: Duplicate import declaration "io.println(int)" at line 5, column 1.
Warning: Duplicate import declaration "io.println(int)" at line 6, column 1.
Warning: Duplicate import declaration "io.println(int)" at line 7, column 1.
Warning: Duplicate import declaration "Quicksort" at line 10, column 1.
Error: Invalid import declaration "Package", constructor cannot be static, at line 12, column 1.
Error: Invalid import declaration "Package(int,boolean)", constructor cannot be static, at line 13, column 1.
Error: Invalid import declaration "Package", constructor cannot return a value, at line 15, column 1.
Error: Invalid import declaration "Package(int,boolean)", constructor cannot return a value, at line 16, column 1.
Error: Duplicate method variable name "a" at line 29, column 35.
Error: Duplicate method variable name "a" at line 29, column 44.
Error: Duplicate method declaration "add(int,boolean,int[])" at line 33, column 16.
Error: Duplicate method declaration "add(int,int)" at line 41, column 16.
Error: Duplicate method declaration "add(int,int)" at line 45, column 20.
Error: Duplicate method declaration "add(int,int)" at line 49, column 18.
